client/network/src/service.rs

```diff
@@ -596,6 +596,12 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkService<B, H> {
 		&self.local_peer_id
 	}
 
+	/// Set reserved peers and reserved_only flag.
+	pub fn set_reserved_peers(&self, peers: HashSet<PeerId>, reserved_only: bool) {
```

Comment:
I would expect this to be two distinct methods like is done on the peerset. E.g. in case someone wants to set `reserved_only` to `true` but does not want to alter the `reserved_peers` themselves.

What do you think?